requirements/customer/scheduling-algorithm.tex

```diff
@@ -64,3 +64,37 @@ \chapter{Scheduling Algorithm}
   However, this behavior changes once the effective priority $p$ of a non-urgent job reaches a certain threshold:
   once the threshold is reached the scheduling algorithm will begin to pause non-urgent running jobs with lower effective priority $p$ if possible and necessary to run the job with an effective priority $p$ above the threshold.
 \end{enumerate}
+
+\begin{figure}
+  \centering
+  \includegraphics[width=0.9\textwidth]{./customer/diagrams/png/main-idea.png}
+  \caption{This diagram depicts one iteration of the scheduling algorithm.
+    The effective priorities of jobs are reevaluated when the set of runnable jobs changes or after a periodic time interval.
+    The scheduling algorithm then attempts to assign jobs to work machines.
+    First it tries to assign all pending urgent jobs, followed by all paused non-urgent jobs, and lastly all pending non-urgent jobs.
+    Inside the aforementioned three steps the scheduling algorithm processes jobs in descending order of their effective priority (high priority first).
+    Diagrams for how the scheduling algorithm decides if a single job can be assigned/resumed are shown in graphics: \hyperref[urgent]{Figure 2} \hyperref[paused]{Figure 3} \hyperref[non-urgent]{Figure 4}}
+\end{figure}
+\begin{figure}
+  \centering
+  \includegraphics[width=\textwidth]{./customer/diagrams/png/urgent-job.png}
```

Comment:
Can we pause some of them to run the urgent one -> Can we pause some of them to run the job